scripts/build-dbt.py

```diff
@@ -0,0 +1,829 @@
+                if len(parts) == 2:
+                    metadata_dict[parts[0]] = parts[1]
+            return metadata_dict
+
+    def package_name(self) -> str:
+        metadata = self.get_metadata()
+        if 'Name' not in metadata:
+            raise ValueError('Wheel with no name?')
+        return metadata['Name']
+
+
+class DockerBuilder:
+    """The docker builder requires the existence of a dbt package"""
+    def __init__(self, dbt_path: Path, version: Version) -> None:
+        self.dbt_path = dbt_path
+        self.version = version
+
+    @property
+    def docker_path(self) -> Path:
+        return self.dbt_path / 'docker'
+
+    @property
+    def dockerfile_name(self) -> str:
+        return f'Dockerfile.{self.version}'
+
+    @property
+    def dockerfile_path(self) -> Path:
+        return self.docker_path / self.dockerfile_name
+
+    @property
+    def requirements_path(self) -> Path:
+        return self.docker_path / 'requirements'
+
+    @property
+    def requirements_file_name(self) -> str:
+        return f'requirements.{self.version}.txt'
+
+    @property
+    def dockerfile_venv_path(self) -> Path:
+        return self.dbt_path / 'build' / 'docker-venv'
+
+    @property
+    def requirements_txt_path(self) -> Path:
+        return self.requirements_path / self.requirements_file_name
+
+    def make_venv(self) -> DistFolderEnv:
+        env = DistFolderEnv(self.dbt_path)
+        env.create(self.dockerfile_venv_path)
+        return env
+
+    def get_frozen(self) -> str:
+        env = self.make_venv()
+        pip_path = self.dockerfile_venv_path / 'bin/pip'
+        cmd = [pip_path, 'freeze']
+        wheel_names = {
+            WheelInfo(wheel_path).package_name() for wheel_path in env.wheels
+        }
+        # remove the dependencies in dbt itself
+        return '\n'.join([
+            dep for dep in collect_output(cmd).split('\n')
+            if dep.split('==')[0] not in wheel_names
+        ])
+
+    def write_lockfile(self):
+        freeze = self.get_frozen()
+        path = self.requirements_txt_path
+        if path.exists():
+            raise ValueError(f'Found existing requirements file at {path}!')
+        os.makedirs(path.parent, exist_ok=True)
+        path.write_text(freeze)
+
+    def get_dockerfile_contents(self):
+        dist_path = (self.dbt_path / 'dist').relative_to(Path.cwd())
+        wheel_paths = ' '.join(
+            os.path.join('.', 'dist', p.name)
+            for p in _require_wheels(self.dbt_path)
+        )
+
+        requirements_path = self.requirements_txt_path.relative_to(Path.cwd())
+
+        return textwrap.dedent(
+            f'''\
+            FROM python:3.8.1-slim-buster
+
+            RUN apt-get update && \
+                apt-get dist-upgrade -y && \
+                apt-get install -y  --no-install-recommends \
+                    netcat curl git ssh  software-properties-common \
+                    make build-essential ca-certificates libpq-dev && \
```

Comment:
does this need netcat / curl / ssh?

Reply:
After a very cursory test, everything appears ok without those!